python/lsst/cp/pipe/ptc.py

```diff
@@ -367,11 +434,269 @@ def runDataRef(self, dataRef, visitPairs):
                 except OperationalError:
                     pass
 
+        if self.config.doCovariancesAstier:
+            # Calculate the full covariances as in Astier+19. The variances for the PTC will be
+            # cov[0,0].
+
+            tupleCovariancesWithTags = self.computeCovariancesAstier(dataRef, visitPairs, detector)
+            # use the np.recarray to obtain the CovFit objects
+            covFits, covFitsNoB = fitData(tupleCovariancesWithTags, maxMu=self.config.maxMeanSignal,
+                                          maxMuElectrons=self.config.maxMeanSignal,
+                                          r=self.config.maximumRangeCovariancesAstier)
+
+            if self.config.makePlots:
+                self.plotCovariancesAstier(dataRef, covFits, covFitsNoB, tupleCovariancesWithTags,
+                                           log=self.log)
+
+            datasetCovAstier = PhotonTransferCurveCovAstierDataset()
+            datasetCovAstier.covariancesTuple = tupleCovariancesWithTags
+            datasetCovAstier.covariancesFits = covFits
+            datasetCovAstier.covariancesFitsWithNoB = covFitsNoB
+
+            dataset = self.getOutputPtcDataCovAstier(covFits)
```

Comment:
Can this generate the thing we want without having to reassign the output values to the PTC dataset?